### DISCUSSION TITLE
Slight style change in history search

### DISCUSSION THREAD

**Discussion by kidonng** [PatrickF1/fzf.fish#242] [Ideas]
Proposing a slight style change in [history search](https://github.com/PatrickF1/fzf.fish/blob/4ebf002121294098629c3618418e6024c564554a/functions/_fzf_search_history.fish#L7-L8):

```diff
-        builtin history --null --show-time="%m-%d %H:%M:%S │ " |
-        _fzf_wrapper --read0 \
+        builtin history --null --show-time=(set_color -o)"%m-%d %H:%M:%S - "(set_color normal) |
+        _fzf_wrapper --read0 --ansi \
```

## Preview

<img width="722" alt="image" src="https://user-images.githubusercontent.com/44045911/171806121-47afd787-7a35-47d8-a08a-2646d83c62bf.png">

## Motivation

- Be consistent with [**Search git log**](https://github.com/PatrickF1/fzf.fish/blob/4ebf002121294098629c3618418e6024c564554a/functions/_fzf_search_git_log.fish#L7)
- While I proposed using box-drawing characters in #167, I didn't find it look nice **as a separator**. To compensate for using dashes (`-`), we are making the timestamp bold to further distinguish it.

## Concern

- **Regarding dashes (`-`)**: this will not hinder search unless you are specifically searching for a single dash, which I honestly have no idea about.
- **Regarding performance**: given [the relatively limited scale fish's history operate on](https://github.com/fish-shell/fish-shell/pull/6493#issuecomment-586359940), ANSI parsing is affordable.

---

P.S. since we are talking about timestamp format, I took the chance to [answer one of your questions](https://github.com/PatrickF1/fzf.fish/discussions/193#discussioncomment-2876509) last year that I forgot about.

**Comment by PatrickF1** [comment]
So you want to go to dashes just because they look nicer?

> Regarding dashes (-): this will not hinder search unless you are specifically searching for a single dash, which I honestly have no idea about.

Good call out. I can see this being an issue because some commands receive std input using -. If they want to search for those commands, it'll be a lot harder to.

> Be consistent with [Search git log](https://github.com/PatrickF1/fzf.fish/blob/4ebf002121294098629c3618418e6024c564554a/functions/_fzf_search_git_log.fish#L7)

That's true but I don't see any value in being consistent with git log. If we really want to, I'd rather change the separator character used in git log to the box drawing vertical line.

**Reply by kidonng** [reply]
> So you want to go to dashes just because they look nicer?

I changed my mind. I think what I really want is more distinct differences between the timestamp and the command, like bold and color (well, if it looks good to you):

<img width="616" alt="image" src="https://user-images.githubusercontent.com/44045911/172060898-477db947-3368-4a5b-ba73-35c340c6d7d0.png">

>> Regarding dashes (-): this will not hinder search unless you are specifically searching for a single dash, which I honestly have no idea about.

> Good call out. I can see this being an issue because some commands receive std input using -. If they want to search for those commands, it'll be a lot harder to.

Actually, even if nothing changes searching for a single dash won't work either, because it will always match the dash in timestamp (which is formatted as `%m-%d %H:%M:%S`). If this sounds like an issue to you we can discuss it elsewhere.

> I'd rather change the separator character used in git log to the box drawing vertical line.

That's a welcoming change, though I would prefer to put it before the commit message, kinda like separate "meta info" from the "message" you are mostly like to search for, just like in command history.

<img width="479" alt="image" src="https://user-images.githubusercontent.com/44045911/172060918-dfc379e6-a47f-4895-9304-3400748df88d.png">

**Comment by PatrickF1** [comment]
> I changed my mind. I think what I really want is more distinct differences between the timestamp and the command, like bold and color (well, if it looks good to you):

Sure I am open to a phab changing the colors or making it bold. I'd be wary about changing the colors though because we can't assume the color scheme the user is using, and it'd be terrible if we hardcoded the same color as their background.

> Actually, even if nothing changes searching for a single dash won't work either, because it will always match the dash in timestamp (which is formatted as %m-%d %H:%M:%S). If this sounds like an issue to you we can discuss it elsewhere.

Good point.

> That's a welcoming change, though I would prefer to put it before the commit message, kinda like separate "meta info" from the "message" you are mostly like to search for, just like in command history.

I'll pass on that change for now.

**Reply by kidonng** [reply]
> I'd be wary about changing the colors though because we can't assume the color scheme the user is using, and it'd be terrible if we hardcoded the same color as their background.

Not when using ANSI colors. And fzf.fish _is_ using colors elsewhere.

**Reply by PatrickF1** [reply]
why don't you think ANSI colors will be a problem in that situation?

**Reply by kidonng** [reply]
Almost anything that doesn't deal with syntax highlight or themes use ANSI colors. And no sane man would set them to be the same or similar to the background (except for black/white, of course).
